safehttp/mux_test.go

```diff
@@ -206,3 +206,271 @@ func TestMuxHandleSameMethodTwice(t *testing.T) {
 
 	mux.Handle("/bar", MethodGet, registeredHandler)
 }
+
```

Comment:
Please test the plugins outside of the `safehttp` package, i.e. in `safehttp_test`. Plugins should not be able to access internals of `safehttp`, only the public APIs.